flutter/lib/ui/painting.dart

```diff
@@ -280,20 +316,101 @@ class Paint {
   ///
   /// This color is not used when compositing. To colorize a layer, use
   /// [colorFilter].
-  Color color = _kDefaultPaintColor;
-  static const Color _kDefaultPaintColor = const Color(0xFF000000);
+  Color get color {
+    final int encoded = _data.getInt32(_kColorOffset, _kFakeHostEndian);
+    return new Color(encoded ^ _kColorDefault);
+  }
+  set color(Color value) {
+    assert(value != null);
```

Comment:
alternatively, maybe we should make null invalid for all the others (except colorFilter). That would solve the current crazy around how null means the opposite of the default for antialiasing, but means the default for the others, and how `p.foo = null; assert(p.foo != null);` asserts for most values of foo here.